lib/mapper.js

```diff
@@ -150,14 +150,24 @@ exports.orderCompleted = function(track, settings){
     };
 
     productPayload.properties = {
-      $event_id: product.productId() || product.id() || track.orderId() + '_' + product.sku(),
       $value: product.price(),
       Name: product.name(),
       Quantity: product.quantity(),
       "Product Categories": [product.category()],
       SKU: product.sku()
     };
 
+    // ensure unique $event_id is associated with each Ordered Product event by combining Order Completed order_id and product's productId or SKU
+    var orderId = payloads.order.properties.$event_id;
+
+    if (product.productId()) {
```

Comment:
Looks like this logic won't work as expected, so reverted. For example, if `product.productId()` is undefined, `undefined` is coerced to a string, which is truthy. So, if `orderId` is 123 and `product.productId() is undefined, `$event_id` would evaluate to "123_undefined"!